### PR TITLE
[FLINK-10704] Fix sql client end to end test failure

### PR DESCRIPTION
**Pull request by yanghua**
## What is the purpose of the change

*This pull request fixes sql client end to end test failure*

## Brief change log

  - *Fix sql client end to end test failure*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

File: flink-end-to-end-tests/test-scripts/common.sh [hunk 1]
```diff
@@ -316,7 +316,7 @@ function check_logs_for_errors {
       | grep -v "java.lang.NoClassDefFoundError: org/apache/hadoop/yarn/exceptions/YarnException" \
       | grep -v "java.lang.NoClassDefFoundError: org/apache/hadoop/conf/Configuration" \
       | grep -v "org.apache.flink.fs.shaded.hadoop3.org.apache.commons.beanutils.FluentPropertyBeanIntrospector  - Error when creating PropertyDescriptor for public final void org.apache.flink.fs.shaded.hadoop3.org.apache.commons.configuration2.AbstractConfiguration.setProperty(java.lang.String,java.lang.Object)! Ignoring this property." \
-      | grep -ic "error")
+      | grep -c "ERROR")
```
Comment (pnowojski) [inline flink-end-to-end-tests/test-scripts/common.sh]: As I have written in the ticket:

> This global `check_logs_for_errors ` is broken design. We might want to have some limited global list of exclusions, but adding more entries to ever growing list is unmaintainable. I think we should filter out this one specific warning in kafka tests and do not mess with changing global rules for all of the tests.

Comment (yanghua) [reply flink-end-to-end-tests/test-scripts/common.sh]: Agree. I thought that you want to filter the warning log globally. Sorry, I misunderstood what you mean.

Comment (pnowojski) [reply flink-end-to-end-tests/test-scripts/common.sh]: Initially I though about ignoring all warnings, since it seemed to me as a better solution compared to changing the behaviour of case sensitiveness globally. But Then I realised that there are already way too many exclusions hardcoded in this file.

Comment (twalthr) [reply flink-end-to-end-tests/test-scripts/common.sh]: @dawidwys I think you are also working on improving the e2e tests here. What is your opinion?

Comment (dawidwys) [reply flink-end-to-end-tests/test-scripts/common.sh]: First of all I agree this is a bit of a broken design, but the current version is that we do not want to introduce a major rework in the bash scripts. The idea right now is that if test requires some custom validation it should disable the default checking and do it in the test itself, so that this decision is a conscious one.

Having that in mind:
- I would definitely leave the case sensitivity as it is
- If you don't want to extend the global list, in other words if you don't want to rely on the default behavior I would disable the log checking via #6959 and add a custom check in the test itself that verifies logs with your extended whitelist (preferably in a trap invoked on exit).

Comment (yanghua) [reply flink-end-to-end-tests/test-scripts/common.sh]: Thanks for your discussion, will try to change the implementation tomorrow.

**Comment by yanghua** [comment]
@pnowojski @twalthr I refer to other end-to-end tests and skip the check.

File: flink-end-to-end-tests/run-pre-commit-tests.sh [hunk 2]
```diff
@@ -48,6 +48,7 @@ echo "Flink distribution directory: $FLINK_DIR"
 # those checks are disabled, one should take care that a proper checks are performed in the tests itself that ensure that the test finished
 # in an expected state.
 
+run_test "SQL Client end-to-end test" "$END_TO_END_DIR/test-scripts/test_sql_client.sh" "skip_check_exceptions"
```
Comment (pnowojski) [inline flink-end-to-end-tests/run-pre-commit-tests.sh]: Quoting @dawidwys:
> If you don't want to extend the global list, (...) I would disable the log checking** via #6959 **_AND_ add a custom check in the test itself that verifies logs with your extended whitelist** (preferably in a trap invoked on exit).

and look two lines above
```
# IMPORTANT:
# (...) Whenever
# those checks are disabled, one should take care that a proper checks are performed in the tests itself that ensure that the test finished
# in an expected state.
```

tbh in this case I would just prefer adding `Error while loading kafka-version.properties :null` to a global exclusion list, instead of copying whole exclusion list to the kafka test.

File: flink-end-to-end-tests/run-nightly-tests.sh [hunk 3]
```diff
@@ -125,7 +125,7 @@ run_test "State TTL RocksDb backend end-to-end test" "$END_TO_END_DIR/test-scrip
 
 run_test "Running Kerberized YARN on Docker test " "$END_TO_END_DIR/test-scripts/test_yarn_kerberos_docker.sh"
 
-run_test "SQL Client end-to-end test" "$END_TO_END_DIR/test-scripts/test_sql_client.sh"
+run_test "SQL Client end-to-end test" "$END_TO_END_DIR/test-scripts/test_sql_client.sh" "skip_check_exceptions"
```
Comment (pnowojski) [inline flink-end-to-end-tests/run-nightly-tests.sh]: Why have you added this test to `run-pre-commit-tests.sh`? If there is some reason behind it, do it in separate commit and remove the runs from `run-nightly-tests.sh`.

Comment (yanghua) [reply flink-end-to-end-tests/run-nightly-tests.sh]: @pnowojski The reason for this is to validate the effect of the change in Travis, because if placed in `run-nightly-tests.sh', it will not be routinely executed. I was going to wait until the PR was approved to withdraw the change, but your suggestion may be more reasonable. I can split it into a separate commit.

File: flink-end-to-end-tests/run-pre-commit-tests.sh (hunk 2 above)
Comment (yanghua) [reply flink-end-to-end-tests/run-pre-commit-tests.sh]: On this point, I think a more reasonable approach is that we can add ignore list or whitelist to the global check method, which may be more scalable. I think this change should be universal and the refactoring should be independent of any specific test. And I saw some tests that skipped the error detection and did not add custom detection. So I chose to skip detection temporarily and reconstruct the global detection method on an independent PR. Finally, we can add this special log to ignore list. What do you think of this idea?

Comment (pnowojski) [reply flink-end-to-end-tests/run-pre-commit-tests.sh]: Please add `Error while loading kafka-version.properties :null` to global ignore list in this PR, to make it fully complete and working. If you want to clean it up later, by moving this `Error while loading kafka-version.properties :null` from global ignore list to a local ignore list, it can be done in separate PR/issue. Merging this with completely disabled exceptions checking is asking for troubles.

**Comment by yanghua** [comment]
@pnowojski updated.

File: flink-end-to-end-tests/test-scripts/elasticsearch-common.sh [hunk 4]
```diff
@@ -87,6 +87,8 @@ function verify_result_hash {
   local numRecords=$3
   local hash=$4
 
+  sleep 20s
```
Comment (yanghua) [inline flink-end-to-end-tests/test-scripts/elasticsearch-common.sh]: @twalthr After many local and Travis tests, I determined that "Executing SQL: Values -> Elasticsearch (upsert)" is very unstable (statistics count results). I think it's because of the prematurely getting results from Elasticsearch. Here I feel that waiting for a while before getting results from Elasticsearch will reduce the chances of failure. cc @pnowojski

Comment (yanghua) [reply flink-end-to-end-tests/test-scripts/elasticsearch-common.sh]: OK, I will create a new issue.

**Comment by yanghua** [comment]
@pnowojski OK, Shall I also remove the commit about "Clearly sleep for a while to wait for the correctness of the elasticsearch statistics." and fix it in a sperate PR?

**Comment by pnowojski** [comment]
You don't have, it can be in this PR - just rename the commit message.

**Comment by yanghua** [comment]
@pnowojski I have updated this commits. Now, FLINK-10604 is only blocked by this PR, I hope it can be merged as soon as possible.

**Comment by yanghua** [comment]
@pnowojski Can we merge this PR?

File: flink-end-to-end-tests/test-scripts/elasticsearch-common.sh (hunk 4 above)
Comment (pnowojski) [reply flink-end-to-end-tests/test-scripts/elasticsearch-common.sh]: Ops, sorry. I misunderstood @twalthr comment. I agree with him that we shouldn't be merging this `sleep 20s` fix, especially that the proper fix should easy (test for the results in a loop). I will look into it.

**Comment by pnowojski** [comment]
I will look into https://issues.apache.org/jira/browse/FLINK-10801 but in the meantime I will just merge 
[FLINK-10704] fix alone.

**Comment by pnowojski** [comment]
Merged without FLINK-10801 fix
- as `af90555fb4` into master
- as `193faaa007` into release-1.7
